src/pastel/Pastel.rei

```diff
@@ -5,6 +5,17 @@
  * LICENSE file in the root directory of this source tree.
  */;
 
+module Make: () => PastelSig.PastelSig;
+module ColorName = ColorName;
+
+type mode =
+  | Terminal
+  | HumanReadable
+  | Disabled;
+
+let getMode: unit => mode;
+let setMode: mode => unit;
```

Comment:
Let's add a `resetMode` as well that sets it to the default. This prevents the need for someone to know `Terminal` is the default. They can pick what mode they want and then just do a reset.

Reply:
I think I like adding resetMode (and just made a commit implementing it), I think in general at the application level people are just going to set a mode and move on.

In library code I can see people wanting to use a different mode for a short period of time, but I think that they actually want to be saving the previous mode from getMode() and restoring it with setMode() rather than calling resetMode(). My concern is that by the existence of resetMode, people will use it in the latter case I alluded to.

Do you have thoughts about that concern? I can certainly think of uses for resetMode, but I think the use cases I outlined in the previous paragraph are going to be the primary ones and I'm a little worried about encouraging the wrong behavior (this is of course somewhat mitigable by documentation, but that approach doesn't really scale)

Reply:
Good point, perhaps a way to address this would be to instead optimize the "use a particular mode for a short time part" not "save and then restore the mode". This can be pretty easy with a 

```
Pastel.useMode(HumanReadable, () => {
  // logic that happens with HumanReadable mode set, and the old mode restored.
  // will have to decide what to do if the current mode changes during the execution of `useMode` though
});
```